packages/cmf/src/cmfConnect.js

```diff
@@ -272,14 +311,63 @@ export default function cmfConnect({
 				}
 			}
 
+			onEvent() {
+				return Object.keys(this.props).reduce(this.onEventDispatch, { toOmit: [] });
+			}
+
+			onEventDispatch(props, key) {
+				if (key.startsWith('on')) {
+					if (key.endsWith('Dispatch')) {
+						props.toOmit.push(key);
+						// eslint-disable-next-line no-param-reassign
+						props[key.replace('Dispatch', '')] = (event, data) => {
+							const payload = Object.assign(
+								{
+									event: serializeEvent(event),
+									data,
+								},
+								this.props[key],
+							);
+							this.props.dispatch(payload);
+						};
+					}
+					if (key.endsWith('ActionCreator')) {
+						props.toOmit.push(key);
+						const newKey = key.replace('ActionCreator', '');
+						let actionCreator = this.props[key];
+						let args;
+						if (typeof this.props[key] === 'object') {
+							actionCreator = this.props[key].id;
+							args = this.props[key].args;
+						}
+						// eslint-disable-next-line no-param-reassign
+						props[newKey] = (event, data) => {
+							if (!args) {
```

Comment:
I'm not sure this ensure that the right arguments are provided to dispatchActionCreator

`dispatchActionCreator(actionCreatorId, event, data, context)`

```
let actionCreator = this.props[key];
let args;
if (typeof this.props[key] === 'object') {
	actionCreator = this.props[key].id;
	args = this.props[key].args; // QUESTION : this is data param, right ? It's from settings so it cannot have event or context, right ? But in your unit test, it's an array, not sure why.
}
if (!args) {
	args = [{ props: this.props, data }];
}
this.dispatchActionCreator(actionCreator, event, args);
```

Reply:
The main idea here is to provide static argument.
For example let say the following action creator has static arguments:
* actions.collections.addOrReplace
* actions.http
* ...
in this use case (static argumetn, it's ok)

Note there is a diff between this.dispatchActionCreator and this.props.dispatchActionCreator.
The first one insert the context for you, so here we don't need to care about.

Then `args` is an array yes, it's like expressions, so it's why it's spread.
If we remove the spread, It forces event, data, context every where. just don't know.